### PR TITLE
Fix typos in validator/

### PR DESCRIPTION
**Pull request by szepeviktor**
@powerivq From #39602

See `typos` config in that PR!

### CODE REVIEW

**Comment by powerivq** [comment]
adding @banaag  for validator review. I do not see problems.

**Review by banaag — APPROVED** [review]
LGTM for validator

**Comment by powerivq** [comment]
Validator test has failed and it needs to be fixed. I think it actually changed the test itself.

**Comment by szepeviktor** [comment]
Please help me where/how to fix it.

File: validator/cpp/htmlparser/data/entities.json [hunk 1]
```diff
@@ -3635,7 +3635,7 @@
     "codepoints": [8911],
     "characters": "\u22CF"
   },
-  "&curren": {
+  "&current": {
```
Comment (powerivq) [inline validator/cpp/htmlparser/data/entities.json:3638]: This is not a typo. Please revert.

File: validator/cpp/htmlparser/parser_test.cc [hunk 2]
```diff
@@ -126,7 +126,7 @@ TEST(ParserTest, LineColTest) {
            }
            default:
              EXPECT_EQ(htmlparser::AtomUtil::ToString(
-                 t.atom) + " was unexecpted.", "");
+                 t.atom) + " was unexpected.", "");
```
Comment (powerivq) [inline validator/cpp/htmlparser/parser_test.cc:129]: Might want to revert these two as well.

File: validator/cpp/htmlparser/testdata/tree-construction/template.dat [hunk 3]
```diff
@@ -1103,7 +1103,7 @@ eof in template
 <body><template><i><menu>Foo</i>
 #errors
 no doctype
-mising /menu
+missing /menu
```
Comment (powerivq) [inline validator/cpp/htmlparser/testdata/tree-construction/template.dat:1106]: And this

File: validator/testdata/feature_tests/ads.html [hunk 4]
```diff
@@ -664,7 +664,7 @@ <h2>Doubleclick no ad</h2>
       data-slot="/4119129/doesnt-exist">
   </amp-ad>
 
-  <h2>Doubleclick with overriden size</h2>
+  <h2>Doubleclick with overridden size</h2>
```
Comment (powerivq) [inline validator/testdata/feature_tests/ads.html:667]: I think it is easier to revert all files in validator/testdata/

**Review by powerivq — CHANGES_REQUESTED** [review]

File: validator/testdata/feature_tests/ads.html (hunk 4 above)
Comment (szepeviktor) [reply validator/testdata/feature_tests/ads.html:667]: Trying with only the above reverts.
Let's see CI!
Revert the whole validator/testdata/ on failure.

File: validator/cpp/htmlparser/data/entities.json (hunk 1 above)
Comment (szepeviktor) [reply validator/cpp/htmlparser/data/entities.json]: Done 🍏

File: validator/cpp/htmlparser/parser_test.cc (hunk 2 above)
Comment (szepeviktor) [reply validator/cpp/htmlparser/parser_test.cc]: Done 🍏 🍏

File: validator/cpp/htmlparser/testdata/tree-construction/template.dat (hunk 3 above)
Comment (szepeviktor) [reply validator/cpp/htmlparser/testdata/tree-construction/template.dat]: Done 🍏

File: validator/testdata/feature_tests/ads.html (hunk 4 above)
Comment (szepeviktor) [reply validator/testdata/feature_tests/ads.html]: Done 🍏

**Review by powerivq — APPROVED** [review]

**Pull request merged**